install/build.py

```diff
@@ -164,13 +164,21 @@ def get_compiler_setting(use_hip):
         else:
             define_macros.append(('CUPY_NO_NVTX', '1'))
 
-    cub_path = os.environ.get('CUB_PATH', '')
-    if os.path.exists(cub_path):
-        # for <cupy/complex.cuh>
-        cupy_header = os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                   '../cupy/core/include')
-        include_dirs.append(cupy_header)
-        include_dirs.append(cub_path)
+    # for <cupy/complex.cuh>
+    cupy_header = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                               '../cupy/core/include')
+    cub_path = None
+    if cuda_path:
+        cub_path = os.path.join(cuda_path, 'include', 'cub')
+        if os.path.exists(cub_path):
+            include_dirs.append(cupy_header)
+        else:
+            cub_path = None
+    if cub_path is None:
+        cub_path = os.environ.get('CUB_PATH', '')
+        if os.path.exists(cub_path):
+            include_dirs.append(cupy_header)
+            include_dirs.append(cub_path)
```

Comment:
Leaving a note: This search precedence is 1. CUDA-bundled CUB (for CUDA 11+), 2 user-provided CUB. This is different from what's currently done in my CUB-bundle PR (#2584). If this PR is merged first, we can move the discussion there.

Reply:
Thank you for pointing that out. I hadn't considered the case of using a user-specified CUB after CUDA 11, but it might be better to allow user-specified CUB to be used in CUDA 11 or later. I will change the search precedence.